source/extensions/tracers/datadog/span.cc

```diff
@@ -0,0 +1,118 @@
+#include "source/extensions/tracers/datadog/span.h"
+
+#include <datadog/dict_writer.h>
+#include <datadog/sampling_priority.h>
+#include <datadog/span_config.h>
+#include <datadog/trace_segment.h>
+
+#include <utility>
+
+#include "source/common/tracing/null_span_impl.h"
+#include "source/extensions/tracers/datadog/time_util.h"
+
+#include "absl/strings/str_cat.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+namespace {
+
+class TraceContextWriter : public datadog::tracing::DictWriter {
+public:
+  explicit TraceContextWriter(Tracing::TraceContext& context) : context_(context) {}
+
+  void set(datadog::tracing::StringView key, datadog::tracing::StringView value) override {
+    context_.setByKey(key, value);
+  }
+
+private:
+  Tracing::TraceContext& context_;
+};
+
+} // namespace
+
+Span::Span(datadog::tracing::Span&& span) : span_(std::move(span)) {}
+
+const datadog::tracing::Optional<datadog::tracing::Span>& Span::impl() const { return span_; }
+
+void Span::setOperation(absl::string_view operation) {
+  if (!span_) {
+    return;
+  }
+
+  span_->set_name(operation);
+}
+
+void Span::setTag(absl::string_view name, absl::string_view value) {
+  if (!span_) {
+    return;
+  }
+
+  span_->set_tag(name, value);
+}
+
+void Span::log(SystemTime, const std::string&) {
+  // Datadog spans don't have in-bound "events" or "logs".
+}
+
+void Span::finishSpan() { span_.reset(); }
```

Comment:
Then datadog span will report the data to collector here automatically?

Reply:
If finishing this span causes the associated [TraceSegment][1] to be finished, then the spans for that segment will be sent to the collector.

[1]: https://github.com/DataDog/dd-trace-cpp/blob/main/src/datadog/trace_segment.h